src/main/scala/com/databricks/labs/overwatch/Optimizer.scala

```diff
@@ -50,27 +50,40 @@ object Optimizer extends SparkSessionWrapper{
 
   /**
    * pass in the overwatch ETL database name to optimize overwatch in parallel
-   * @param args ["overwatch_etl"]
+   * @param args(0)   Overwatch ETL Database.
+   * @param args(1)   Organization_ID for which optmization need to be performed.(Optional). By default optimizer will run for all
+   *                  the organization ID in pipeline_report.
    */
   def main(args: Array[String]): Unit = {
 
-    val overwatchETLDB = if (args.length == 1) {
-      args(0)
+
+    val (overwatchETLDB, orgIDs) = if (args.length == 1) {
+      val orgIdList = spark.table(s"${args(0)}.pipeline_report").select("organization_id").distinct().collect().map(x => x(0).toString)
+      (args(0), orgIdList)
+    } else if (args.length == 2) {
+      val cntOrgID = spark.table(s"${args(0)}.pipeline_report").select("organization_id").filter('organization_id === args(1)).distinct.count()
+      if (cntOrgID > 0) {
+        (args(0), Array(args(1)))
+      } else {
+        throw new BadConfigException("Input Organization_ID is not part of the Overwatch Deployment for which you want to run the optimizer")
+      }
     } else {
-      throw new BadConfigException(s"Main class requires at least 1 but less than 5 arguments. Received ${args.length} " +
+      throw new BadConfigException(s"Main class requires at least 1 but less than 3 arguments. Received ${args.length} " +
         s"arguments. Please review the docs to compose the input arguments appropriately.")
     }
 
-    val workspace = getLatestWorkspace(overwatchETLDB)
-    val config = workspace.getConfig
-    if (config.debugFlag) println(JsonUtils.objToJson(config.inputConfig).compactString)
-    val bronze = Bronze(workspace, suppressReport = true, suppressStaticDatasets = true)
-    val silver = Silver(workspace, suppressReport = true, suppressStaticDatasets = true)
-    val gold = Gold(workspace, suppressReport = true, suppressStaticDatasets = true)
+    orgIDs.foreach { orgId =>
```

Comment:
I don't think this is quite right, We should have a path for either optimizing a single orgID (debatable how many times this will be used), and another path for optimizing the entire overwatch_etl DB. 
The way this is written right now we'll run the same optimization/vacuum steps for all tables, one partition at a time, which will be very inefficient.